core/src/main/java/org/kohsuke/stapler/export/ClassAttributeBehaviour.java

```diff
@@ -0,0 +1,103 @@
+package org.kohsuke.stapler.export;
+
+import org.jvnet.tiger_types.Types;
+
+import java.lang.reflect.Type;
+
+/**
+ * Setting that controls how the '_class' attribute will be produced in the output.
+ *
+ * <p>
+ * Three basic constants are defined:
+ *
+ * <dl>
+ *     <dt>{@link #NONE}
+ *     <dd>No type information will show up in the output whatsoever
+ *
+ *     <dt>{@link #ALWAYS}
+ *     <dd>Every object gets the type information explicitly written out
+ *
+ *     <dt>{@link #IF_NEEDED}
+ *     <dd>Type information will be produced only when it is necessary,
+ *         for example when the declared type and the actual type differ.
+ * </dl>
+ *
+ * <p>
+ * In addition, {@link #simple()} can be used to produce just the simple name of the class,
+ * whereas by default {@linkplain Class#getName() full class name} will be printed.
```

Comment:
BTW `DescribableHelper` is designed to require/emit a FQN iff there is ambiguity in `simpleName`s amongst implementations of a given extension point. I think that is not easily done here, since Stapler does not have access to extension lists.